src/index.ts

```diff
@@ -111,14 +111,22 @@ function logTracingHeader(response: Response) {
   log.info(`Request tracing with traceID=${response.headers.get("x-web3-correlation-id")}`);
 }
 
-export const promiseTimeout = <T>(ms: number, promise: Promise<T>): Promise<T> => {
+export const promiseTimeout = async <T>(ms: number, promise: Promise<T>): Promise<T> => {
+  let timeoutFunc: ReturnType<typeof setTimeout> | null = null;
+
   const timeout = new Promise<T>((_resolve, reject) => {
-    const id = setTimeout(() => {
-      clearTimeout(id);
+    timeoutFunc = setTimeout(() => {
       reject(new Error(`Timed out in ${ms}ms`));
     }, ms);
   });
-  return Promise.race<T>([promise, timeout]);
+
+  const result = await Promise.race<T>([promise, timeout]);
```

Comment:
if the original promise rejects, Promise.race throws and timeout is not cleared.